packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -201,16 +201,36 @@ export class GarbageCollector implements IGarbageCollector {
 				try {
 					// For newer documents, GC data should be present in the GC tree in the root of the snapshot.
 					const gcSnapshotTree = baseSnapshot.trees[gcTreeKey];
-					if (gcSnapshotTree !== undefined) {
-						return getGCDataFromSnapshot(gcSnapshotTree, readAndParseBlob);
+					if (gcSnapshotTree === undefined) {
+						// back-compat - Older documents will have the GC blobs in each data store's snapshot tree.
+						return getSnapshotDataFromOldSnapshotFormat(
+							baseSnapshot,
+							createParams.metadata,
+							readAndParseBlob,
+						);
 					}
 
-					// back-compat - Older documents will have the GC blobs in each data store's snapshot tree.
-					return getSnapshotDataFromOldSnapshotFormat(
-						baseSnapshot,
-						createParams.metadata,
+					const snapshotData = await getGCDataFromSnapshot(
+						gcSnapshotTree,
 						readAndParseBlob,
 					);
+
+					if (
```

Comment:
Subjective nit: I would flip this `if` condition and bring the comment up here, so then the bottom of the function is the "happy path" and exceptional cases (like the back compat one above and this one) are called out earlier on.